### ISSUE TITLE
Learning Rate stays at 0.0?

### ISSUE THREAD

**Issue by solarslurpi** [huggingface/autotrain-advanced#268]
- The training loss is consistently is 0.0 - I'm messing something up either in the parameters or because of the model I am trying to fine tune. I started with my own dataset and then thought perhaps there was a problem in the dataset format (there is one column, text with instruction/response).  `--data-path tatsu-lab/alpaca ` dataset and had the same challenge - no training loss.  What is missing? Thank you.
```
--text-column text \
--lr ${LEARNING_RATE} \
--batch-size ${BATCH_SIZE} \
--epochs ${NUM_EPOCHS} \
--block-size ${BLOCK_SIZE} \
--warmup-ratio ${WARMUP_RATIO} \
--lora-r ${LORA_R} \
--lora-alpha ${LORA_ALPHA} \
--lora-dropout ${LORA_DROPOUT} \
--weight-decay ${WEIGHT_DECAY} \
--gradient-accumulation ${GRADIENT_ACCUMULATION} \
$( [[ "$USE_FP16" == "True" ]] && echo "--fp16" ) \
$( [[ "$USE_PEFT" == "True" ]] && echo "--use-peft" ) \
$( [[ "$USE_INT4" == "True" ]] && echo "--use-int4" ) 
# $( [[ "$PUSH_TO_HUB" == "True" ]] && echo "--push-to-hub --token ${HF_TOKEN} --repo-id ${REPO_ID}" )
```
Here is my input (NOTE: the model was picked because of its small size):
```
project_name = 'test_tune' # @param {type:"string"}
model_name = 'EleutherAI/pythia-70m' # @param {type:"string"}



#@markdown ---
#@markdown #### Hyperparameters
learning_rate = 2e-4 # @param {type:"number"}
num_epochs = 2  #@param {type:"number"}
batch_size = 1 # @param {type:"slider", min:1, max:32, step:1}
block_size = 1024 # @param {type:"number"}
trainer = "sft" # @param ["default", "sft"] {type:"raw"}
warmup_ratio = 0.1 # @param {type:"number"}
weight_decay = 0.01 # @param {type:"number"}
gradient_accumulation = 4 # @param {type:"number"}
use_fp16 = True # @param ["False", "True"] {type:"raw"}
use_peft = True # @param ["False", "True"] {type:"raw"}
use_int4 = True # @param ["False", "True"] {type:"raw"}
lora_r = 16 #@param {type:"number"}
lora_alpha = 32 #@param {type:"number"}
lora_dropout = 0.05 #@param {type:"number"}
```
and the output:
```
> INFO    Running LLM
> INFO    Params: Namespace(version=False, train=True, deploy=False, inference=False, data_path='tatsu-lab/alpaca', train_split='train', valid_split=None, text_column='text', model='EleutherAI/pythia-70m', learning_rate=0.0002, num_train_epochs=2, train_batch_size=1, warmup_ratio=0.1, gradient_accumulation_steps=4, optimizer='adamw_torch', scheduler='linear', weight_decay=0.01, max_grad_norm=1.0, seed=42, add_eos_token=False, block_size=1024, use_peft=True, lora_r=16, lora_alpha=32, lora_dropout=0.05, logging_steps=-1, project_name='test_tune', evaluation_strategy='epoch', save_total_limit=1, save_strategy='epoch', auto_find_batch_size=False, fp16=True, push_to_hub=False, use_int8=False, model_max_length=1024, repo_id=None, use_int4=True, trainer='default', target_modules=None, merge_adapter=False, token=None, backend='default', username=None, func=<function run_llm_command_factory at 0x7d4fc9b775b0>)
...
Extracting data files: 100% 1/1 [00:00<00:00, 1167.35it/s]
Generating train split: 52002 examples [00:00, 313492.65 examples/s]
Using pad_token, but it is not set yet.
You are resizing the embedding layer without providing a `pad_to_multiple_of` parameter. This means that the new embedding dimension will be 50277. This might induce some performance reduction as *Tensor Cores* will not be available. For more details about this, or help on choosing the correct value for resizing, refer to this guide: https://docs.nvidia.com/deeplearning/performance/dl-performance-matrix-multiplication/index.html#requirements-tc
Running tokenizer on train dataset:  12% 6000/52002 [00:01<00:13, 3400.38 examples/s]Token indices sequence length is longer than the specified maximum sequence length for this model (1034 > 1024). Running this sequence through the model will result in indexing errors
Running tokenizer on train dataset: 100% 52002/52002 [00:19<00:00, 2604.23 examples/s]
Grouping texts in chunks of 1024 (num_proc=4): 100% 52002/52002 [00:10<00:00, 4756.07 examples/s]
> INFO    creating trainer
{'loss': 0.0, 'learning_rate': 0.0, 'epoch': 0.8}
{'loss': 0.0, 'learning_rate': 0.0, 'epoch': 1.6}
{'train_runtime': 1085.9721, 'train_samples_per_second': 10.315, 'train_steps_per_second': 2.578, 'train_loss': 0.0, 'epoch': 2.0}
100% 2800/2800 [18:05<00:00,  2.58it/s]
> INFO    Finished training, saving model...
```

**Comment by abhishekkrthakur** [comment]
you should probably disable peft with pythia 70m
